pkg/kubelet/cm/deviceplugin/pod_devices.go

```diff
@@ -17,207 +17,346 @@ limitations under the License.
 package deviceplugin
 
 import (
+	"sync"
+
 	"github.com/golang/glog"
 
+	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
-type deviceAllocateInfo struct {
-	// deviceIds contains device Ids allocated to this container for the given resourceName.
-	deviceIds sets.String
+type devAllocation interface {
+	// get a deep copy of all maps managed.
+	allocations() map[string]containerMap
+
+	// get the set of pod IDs known to the device manager
+	pods() sets.String
+
+	// Check if the specified pod uid is known to have allocations
+	hasPod(pod string) bool
+
+	// Enumerate all devices known to be already allocated by returning a map
+	// keyed by resource name.
+	devices() map[string]sets.String
+
+	// Enumerate all known device allocations for the given container on
+	// specified resource.
+	devicesByContainer(pod, container, resource string) sets.String
+
+	// Insert new allocation record to the registry.
+	insert(pod, container, resource string, devices sets.String, resp *pluginapi.AllocateResponse)
+
+	// Update the allocation registry based on the provided list of active
+	// pods. Records for inactive pods are pruned.
+	update(activePods []*v1.Pod) map[string]sets.String
+
+	// Marshal the registry into data for checkpointing
+	toCheckpointData() []podDevicesCheckpointEntry
+
+	// Unmarshal checkpointed data into registry
+	fromCheckpointData(data []podDevicesCheckpointEntry)
+
+	// Collect device information into container runtime settings.
+	collectRunOptions(pod, container string) *DeviceRunContainerOptions
+}
+
+type deviceInfo struct {
+	// ids contains device Ids allocated to this container for the given resourceName.
```

Comment:
should we call it `resource Ids` instead? Because we are using `Devices` for device files in `AllocateResponse`.
Similarly at all other instances as well.